File: localbite-backend/app/crud/menu.py
from sqlalchemy.orm import Session
from app.models.menu import MenuItem
from app.schemas.menu import MenuItemCreate, MenuItemBase

def create(db: Session, payload: MenuItemCreate) -> MenuItem:
    db_obj = MenuItem(
        restaurant_id=payload.restaurant_id,
        item_name=payload.item_name,
        price=payload.price,
        availability=payload.availability,
        category=payload.category
    )
    db.add(db_obj)
    db.commit()
    db.refresh(db_obj)
    return db_obj

def get_by_id(db: Session, menu_id: int) -> MenuItem | None:
    return db.query(MenuItem).filter(MenuItem.menu_id == menu_id).first()

def list_by_restaurant(db: Session, restaurant_id: int, skip: int = 0, limit: int = 100) -> list[MenuItem]:
    return db.query(MenuItem).filter(MenuItem.restaurant_id == restaurant_id).offset(skip).limit(limit).all()

def update(db: Session, db_obj: MenuItem, payload: MenuItemBase) -> MenuItem:
    updates = payload.model_dump(exclude_unset=True)
    for field, value in updates.items():
        setattr(db_obj, field, value)
    db.add(db_obj)
    db.commit()
    db.refresh(db_obj)
    return db_obj

def delete(db: Session, db_obj: MenuItem) -> None:
    db.delete(db_obj)
    db.commit()

File: localbite-frontend/lib/api/menu.ts
const API_URL = "http://172.26.56.184:8000/api/v1";

export interface MenuItem {
  menu_id: number;
  restaurant_id: number;
  item_name: string;
  price: number;
  category?: string;
  availability: boolean;
}

export async function getMenuByRestaurant(
  restaurantId: number,
  skip: number = 0,
  limit: number = 100
): Promise<MenuItem[]> {
  try {
    const response = await fetch(
      `${API_URL}/menu/restaurant/${restaurantId}?skip=${skip}&limit=${limit}`,
      {
        credentials: "include",
      }
    );

    if (!response.ok) {
      throw new Error("Failed to fetch menu items");
    }

    const data: MenuItem[] = await response.json();
    return data.filter((item) => item.availability);
  } catch (error) {
    if (error instanceof Error) {
      throw new Error(`Failed to fetch menu: ${error.message}`);
    }
    throw new Error("Failed to fetch menu");
  }
}

export async function getMenuItem(menuId: number): Promise<MenuItem> {
  try {
    const response = await fetch(`${API_URL}/menu/${menuId}`, {
      credentials: "include",
    });

    if (!response.ok) {
      throw new Error("Menu item not found");
    }

    const data: MenuItem = await response.json();
    return data;
  } catch (error) {
    if (error instanceof Error) {
      throw new Error(`Failed to fetch menu item: ${error.message}`);
    }
    throw new Error("Failed to fetch menu item");
  }
}
